### PR TITLE
Add a definition for grpc_global_wakeup_fd in ev_posix.c

### PR DESCRIPTION
**Pull request by ncteisen**
This fixes the linker error mentioned in #8766. Running make on MacOS will still not build, but now the problem lies in recent changes to Cronet. This PR will unblock @gcasto from finishing his Cronet gRPC fix discussed in #8684.

This fix does allow clients and servers that don't use cronet (like make interop_server and make interop_client) to be built (this was not the case before, since the linker error broke everything).

This change does not affect the status of building on Linux at all.

### CODE REVIEW

**Comment by ncteisen** [comment]
A little more context. This was broken in #8680. MacOS's extern grpc_global_wakeup_fd previously linked to [this line](https://github.com/grpc/grpc/pull/8680/files#diff-9f881f5af5d224c806ddd2701123017cL741)

File: src/core/lib/iomgr/ev_posix.c [hunk 1]
```diff
@@ -51,6 +51,7 @@
 /** Default poll() function - a pointer so that it can be overridden by some
  *  tests */
 grpc_poll_function_type grpc_poll_function = poll;
+grpc_wakeup_fd grpc_global_wakeup_fd;
```
Comment (sreecha) [inline src/core/lib/iomgr/ev_posix.c]: We should remove `grpc_global_wakeup_fd` definition from `ev_epoll_linux.c`

**Comment by ncteisen** [comment]
Cherry picked some changes from #8684, and now MacOS builds with no errors

**Comment by muxi** [comment]
Thank you Noah. We are in need of this fix here at mobile :)

**Comment by ncteisen** [comment]
Merged in #8862 since those changes clashed with mine. This fix is still needed to get the build working on MacOS again

**Comment by ncteisen** [comment]
Moved the relevant changes to #8869. This can now be closed

**Pull request closed without merging**